package-res/resources/web/pentaho/visual/ccc/abstract/View.js

```diff
@@ -1023,12 +1026,27 @@ define([
       // Possible to create colorMap based on memberPalette?
       if(C || M) {
         if(!C || M > 1) {
-          // Use measure colors.
+          colorMap = memberPalette["[Measures].[MeasuresLevel]"];
+
+          // Use measure colors
+          // a) C == 0 && M > 0
+          // b) C >  0 && M > 1
+
           // More than one measure MappingAttributeInfo or no color MappingAttributeInfos.
           // var keyIncludesMeasureDiscriminator = M > 1 && C > 0;
           // When the measure discriminator exists, it is the last MappingAttributeInfo.
-          colorMap = this._copyColorMap(null, memberPalette["[Measures].[MeasuresLevel]"]);
+
+          if(M === 1) {
```

Comment:
This was the condition missing. The remaining changes to the code are to be able to return a final color scale, instead of a colorMap object in this case.